Assets/Scripts/Game/UserInterfaceWindows/DaggerfallInventoryWindow.cs

```diff
@@ -908,16 +940,80 @@ protected void AddLocalItem(DaggerfallUnityItem item)
         /// Creates filtered list of remote items.
         /// For now this just creates a flat list, as that is Daggerfall's behaviour.
         /// </summary>
-        protected virtual void FilterRemoteItems()
+        protected virtual void FilterRemoteItems(bool applyExtraFilter = true)
         {
+            DaggerfallUnityItem item;
             // Clear current references
             remoteItemsFiltered.Clear();
 
             // Add items to list
             if (remoteItems != null)
                 for (int i = 0; i < remoteItems.Count; i++)
-                    remoteItemsFiltered.Add(remoteItems.GetItem(i));
+                {
+                    if (applyExtraFilter)
+                    {
+                        item = remoteItems.GetItem(i);
+                        if (ItemPassesFilter(item))
+                            remoteItemsFiltered.Add(item);
+
+                    }
+                    else
+                        remoteItemsFiltered.Add(remoteItems.GetItem(i));
+                }
         }
+        ///<summary>
+        ///Checks whether inventory item meets criteria of filter
+        ///</summary>
+        protected virtual bool ItemPassesFilter(DaggerfallUnityItem item)
+        {
+            bool itemPasses = true;
+            bool iterationPass = false;
+
+            if (filterString.Length == 0)
+                return true;
+
+            foreach (string word in filterString.Split(' '))
+            {
+                if (word.Trim().Length > 0)
+                {
+                    if (word[0] == '-')
+                    {
+                        iterationPass = true;
+                        if (item.LongName.ToLower().Contains(word.Remove(0, 1)))
+                            iterationPass = false;
+                        else if (itemGroupNames[(int)item.ItemGroup].Contains(word.Remove(0, 1)))
+                            iterationPass = false;
+                    }
+                    else
+                    {
+                        iterationPass = false;
+                        if (item.LongName.ToLower().Contains(word))
+                            iterationPass = true;
+                        else if (itemGroupNames[(int)item.ItemGroup].Contains(word))
+                            iterationPass = true;
+                    }
+
+                    if (iterationPass == true && itemPasses == true)
+                        itemPasses = true;
+                    else
+                    {
+                        itemPasses = false;
+                        break;
+                    }
```

Comment:
This can be simplified

```cs
if (!iterationPass)
{
    itemPasses = false;
    break;
}

```

or even

```cs
if (!iterationPass)
    return false;
```